### ISSUE TITLE
reload() puts comman name after command line arguments

### ISSUE THREAD

**Issue by a1s** [pallets/quart#328]
When Quart development server is run in auto-reload mode from a Python script, `reload()` puts the script name at the end of the command line.   So the command line arguments are passed to the Python interpreter and not to the application script.

Example script:

```python
import sys

import click
from quart import Quart

app = Quart(__file__)

@click.command()
@click.option("-c", "--config")
def run(config):
    click.echo(sys.argv)
    app.run(debug=True)

if __name__ == "__main__":
    run()
```

When this script is run with option `-c whatever`, auto-reload fails:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'whatever' is not defined
```

Environment:

- Python version: 3.12.2
- Quart version: 0.19.4

**Comment by a1s** [comment]
This fixes the error: [reload.patch](https://github.com/pallets/quart/files/14693019/reload.patch)

**Comment by pgjones** [comment]
Thanks fixed in b2b7d48ad35a227dbd14ad6a6be694e595d215aa
